frontend/containers/forms/uploader/component.tsx

```diff
@@ -0,0 +1,183 @@
+import { useEffect, useState, useCallback } from 'react';
+
+import { FileRejection, useDropzone } from 'react-dropzone';
+import { FieldValues, Path } from 'react-hook-form';
+import { FormattedMessage } from 'react-intl';
+
+import classNames from 'classnames';
+
+import Image from 'next/image';
+
+import { ProjectImageGallery } from 'types/project';
+
+import { directUpload } from 'services/direct-upload/directUpload';
+
+import { UploaderProps } from './types';
+
+export const bytesToMegabytes = (bytes: number): number => {
+  return bytes / 1000000;
+};
+
+export const FILE_UPLOADER_MAX_SIZE = 1500000;
+
+export const Uploader = <FormValues extends FieldValues>({
+  fileTypes,
+  maxFiles = 1,
+  maxSize = FILE_UPLOADER_MAX_SIZE,
+  disabled = false,
```

Comment:
This should come from `registerOptions`. It also includes other parameters such as `required`.

Reply:
You haven't removed `disabled` from here. You should read it from `registerOptions.disabled`. Don't forget to update `types.ts`.